### DISCUSSION TITLE
Support for Vive Focus Vision on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by Leseratte10** [alvr-org/ALVR#2475] [Q&A]
I just got the Vive Focus Vision and wanted to test this tool to run PC VR games wirelessly on my headset, but I've ran into two (seperate) issues: 

- It looks like support needs to be added for each particular VR headset, and the Vive Focus Vision isn't mentioned in the readme (only the Focus 3 / XR are), nor is it mentioned in the `alvr/client_core/src/platform/mod.rs` file. Is there any data / debug log / etc. I can make available from my headset to help adding support? Is that even needed or should this work without explicit support? I would just test it myself but I ran into a 2nd issue: 

- Both the alvr_dashboard and the alvr launcher just display error messages about not finding functions from glibc 2.35 and below. I am running Ubuntu 20.04.6 which comes with glibc 2.31. It looks like the Linux binaries are being compiled on the "ubuntu-latest" Github CI platform, which currently is Ubuntu 22.04 and will resolve to Ubuntu 24.04 very soon. This means the released binaries can only be used by people running Ubuntu 22.04/24.04 or newer. 

A) is there a particular reason for that (features needed on newer distributions) or would it be possible to lower the version used for the packaging so the tool can run on older systems as well? (If there's no actual dependencies to newer OSes it should be enough to replace "ubuntu-latest" with "ubuntu-20.04"). Or do I have to resort to building from source code? I haven't tried doing that yet because I didn't want to spend a bunch of time trying to get it to compile when I don't know if it'll even work with my VR headset ...

**Comment by zmerp** [comment]
I can speak about the headset, i have reports it works. We don't have a device name yet so it will show as "HTC Vive (Unknown)"

**Reply by zmerp** [reply]
if you can, could you run this (with Focus Vision conneected via USB):
```
adb devices -l
```
this will give me the info necessary to add the display name support

**Reply by Leseratte10** [reply]
Sure, thanks for your answer: 

`device usb:2-5 product:kona model:VIVE_Focus_Vision device:kona transport_id:1`

**Reply by Leseratte10** [reply]
By the way, I did finally manage to compile ALVR on Ubuntu 20.04 but SteamVR itself seems to run into all kinds of issues. I guess I'll try again when I'm on a newer OS.

**Reply by zmerp** [reply]
Thanks!

**Comment by Entity325** [comment]
I can confirm that ALVR works with Vive Focus Vision on Arch Linux. I don't know how different that would be for a Debian-based distro.

I am currently running it using the Quest 2 profile, and everything is "close enough," except the grip angle is 20-30 degrees too far forward, which makes aiming in Pavlov feel really weird.
